File: fyp_be/src/main/java/com/example/fyp_be/model/EventFeedback.java
package com.example.fyp_be.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Data
@Entity

@Table(name = "event_feedback")
public class EventFeedback {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer feedbackId;

    @ManyToOne
    @JoinColumn(name = "event_id", nullable = false)
    private Event event;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private int rating; // e.g., 1 to 5 stars

    @Column(length = 1000) // Allow for a decent length message
    private String comment;

    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    // Getters, Setters, etc.
}

File: fyp_fe_androidNative2/app/src/main/java/com/navigine/indoornavigationdemo/data/EventFeedback.kt
package com.navigine.indoornavigationdemo.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class EventFeedback(
    @Json(name = "feedbackId")
    val feedbackId: Int,
    @Json(name = "rating")
    val rating: Int,
    @Json(name = "comment")
    val comment: String?,
    @Json(name = "createdAt")
    val createdAt: String
)